src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -106,7 +111,7 @@ public void save(AddressBook addressBook) throws StorageOperationException {
             marshaller.marshal(toSave, fileWriter);
 
         } catch (IOException ioe) {
-            throw new StorageOperationException("Error writing to file: " + path);
+            throw new StorageReadOnlyException("Warning: Storage file is read only. Any changes will not be saved");
```

Comment:
Good effort in trying to handle the exception =).
Recall from tutorial other ways of handle this exception.